Make it work when linking with -Wl,--as-needed
See http://www.gentoo.org/proj/en/qa/asneeded.xml
bicatali@gentoo.org - 06/2013

diff -Nur astrometry.net-0.43.orig/blind/Makefile astrometry.net-0.43/blind/Makefile
--- astrometry.net-0.43.orig/blind/Makefile	2013-03-01 12:29:34.000000000 -0800
+++ astrometry.net-0.43/blind/Makefile	2013-06-13 15:54:15.915430597 -0700
@@ -140,7 +140,7 @@
 
 $(BACKEND_SO): $(BACKEND_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(LIBKD_LIB) \
 		$(QFITS_LIB) $(GSL_LIB)
-	$(CC) $(SHAREDLIBFLAGS) -o $@ $^
+	$(CC) $(SHAREDLIBFLAGS) $^ -o $@
 
 # old and miscellaneous executables that aren't part of the pipeline.
 OLDEXECS := plotquads rawstartree checkquads
@@ -350,7 +350,7 @@
 
 demo_dsmooth: demo_dsmooth.o dsmooth.o \
 	$(COMMON)/cairoutils.o $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) 
-	$(CC) -o $@ $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) -pg $(LDLIBS)
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) -pg $(LDLIBS) -o $@
 
 control-program: control-program.o \
 		$(BACKEND_OBJS) $(SIMPLEXY_OBJ) \
@@ -359,7 +359,7 @@
 
 # qfits is only needed for writing debugging images...
 image2xy: image2xy-main.o image2xy-files.o $(SIMPLEXY_OBJ) $(CFITS_DEP) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(CFITS_LIB) $(LDLIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(CFITS_LIB) $(LDLIBS) -o $@
 
 ALL_OBJ += image2xy-main.o image2xy-files.o
 
@@ -397,12 +397,12 @@
 
 wcs-xy2rd: wcs-xy2rd-main.o wcs-xy2rd.o $(UTIL_OBJS) \
 		$(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) $(WCS_LIB_DEP)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(WCS_LIB) $(LDLIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCS_LIB) $(LDLIBS) -o $@
 ALL_OBJ += wcs-xy2rd-main.o
 
 wcs-rd2xy: wcs-rd2xy-main.o wcs-rd2xy.o $(UTIL_OBJS) \
 		$(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) $(WCS_LIB_DEP)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(WCS_LIB) $(LDLIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCS_LIB) $(LDLIBS) -o $@
 ALL_OBJ += wcs-rd2xy-main.o
 
 wcs-pv2sip: wcs-pv2sip.o tweak.o \
@@ -412,10 +412,10 @@
 hpgrid: hpgrid.o $(ANUTILS_LIB) $(GSL_LIB)
 
 $(CFITS_UTILS) :: %: %.o $(CFITS_DEP)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(CFITS_LIB) $(LDLIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(CFITS_LIB) $(LDLIBS) -o $@
 
 fitsverify: ftverify.c fvrf_data.c fvrf_file.c fvrf_head.c fvrf_key.c fvrf_misc.c $(CFITS_DEP)
-	$(CC) -DSTANDALONE -trigraphs $(CFITS_INC) -o $@ $^ $(CFITS_LIB) -lm
+	$(CC) -DSTANDALONE -trigraphs $(CFITS_INC) $^ $(CFITS_LIB) -lm -o $@
 
 $(OLDEXECS) :: %: %.o $(OLDEXECS_OBJS) \
 		$(ANUTILS_LIB) $(GSL_LIB) $(LIBKD_LIB) $(QFITS_LIB)
@@ -427,12 +427,12 @@
 
 ucac3tofits: ucac3tofits.o ucac3-fits.o ucac3.o \
 		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) -o $@ $^ $(LDFLAGS) -lbz2 $(LDLIBS)
+	$(CC) $^ $(LDFLAGS) -lbz2 $(LDLIBS) -o $@
 
 read_nomad: read_nomad.o nomad.o
 
 2masstofits: $(2MASSOBJ) $(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(ZLIB_LIB) $(LDLIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(ZLIB_LIB) $(LDLIBS) -o $@
 
 diffractionFlag_check: diffractionFlag_check.o usnob-fits.o usnob.o \
 		$(UTIL_OBJS) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
@@ -496,7 +496,7 @@
 		hpquads.o uniformize-catalog.o merge-index.o unpermute-quads.o unpermute-stars.o \
 		quad-builder.o quad-utils.o codefile.o wcs-xy2rd.o \
 		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(WCSLIB_LIB) $(LDLIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCSLIB_LIB) $(LDLIBS) -o $@
 ALL_OBJ += local-index.o
 
 index-to-table: index-to-table.o \
@@ -515,14 +515,14 @@
 
 verify: verify-main.o verify.o verify2.o matchfile.o matchobj.o $(PLOTSTUFF) $(COMMON)/cairoutils.o \
 		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) $(GSL_LIB)
-	$(CC) -o $@ $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB)
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) -o $@
 
 keirthing: keirthing.o tweak.o \
 		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(GSL_LIB)
 
 dstnthing: dstnthing.o tweak2.o tweak.o matchfile.o matchobj.o verify.o scamp.o $(PLOTSTUFF) \
 		$(COMMON)/cairoutils.o $(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) $(GSL_LIB) 
-	$(CC) -o $@ $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LDLIBS)
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LDLIBS) -o $@
 ALL_OBJ += dstnthing.o
 
 ALL_OBJ += verify-main.o
@@ -532,14 +532,14 @@
 		fits-guess-scale.o wcs-rd2xy.o new-wcs.o \
 		scamp.o \
 		$(SIMPLEXY_OBJ) $(CFITS_DEP) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(WCSLIB_LIB) $(CFITS_LIB) $(LDLIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCSLIB_LIB) $(CFITS_LIB) $(LDLIBS) -o $@
 
 ALL_OBJ += scamp.o cut-table.o
 
 augment-xylist: augment-xylist-main.o augment-xylist.o tabsort.o cut-table.o fits-guess-scale.o \
 		image2xy-files.o resort-xylist.o blindutils.o \
 		$(SIMPLEXY_OBJ) $(ANFILES_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(CFITS_DEP)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(CFITS_LIB) $(LDLIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(CFITS_LIB) $(LDLIBS) -o $@
 
 ALL_OBJ += augment-xylist-main.o
 
@@ -548,20 +548,20 @@
 		tweak.o \
 		$(UTIL_OBJS) \
 		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) $(GSL_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@.tmp $^ $(LDLIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(LDLIBS) -o $@.tmp
 	mv $@.tmp $@
 
 tweak: tweak-main.o verify.o matchfile.o \
 		$(COMMON)/cairoutils.o $(PLOTSTUFF) \
 		$(UTIL_OBJS) \
 		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) $(GSL_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LDLIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LDLIBS) -o $@
 
 whynot: whynot.o matchfile.o matchobj.o codefile.o solvedclient.o \
 		solvedfile.o verify.o quad-utils.o \
 		$(PLOTSTUFF) $(COMMON)/cairoutils.o $(CATS_LIB) \
 		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) $(GSL_LIB) 
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LDLIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LDLIBS) -o $@
 ALL_OBJ += whynot.o
 
 quadcenters: quadcenters.o \
@@ -597,7 +597,7 @@
 
 solvedserver: solvedserver.o solvedfile.o $(UTIL_OBJS) $(ANUTILS_LIB) $(GSL_LIB) \
 		$(QFITS_LIB)
-	$(CC) $(CFLAGS) -o $@ $^ $(LDLIBS)
+	$(CC) $(CFLAGS) $^ $(LDLIBS) -o $@
 
 plotcat: plotcat.o 2mass-fits.o 2mass.o nomad-fits.o nomad.o catalog.o usnob-fits.o usnob.o \
 		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(LIBKD_LIB) \
@@ -610,28 +610,28 @@
 	$(MAKE) -C $(COMMON) cairoutils.o
 
 onepixel: onepixel.o $(COMMON)/cairoutils.o
-	$(CC) -o $@ $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(LDLIBS)
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(LDLIBS) -o $@
 
 plotquad: plotquad.o matchfile.o matchobj.o $(COMMON)/cairoutils.o \
 		$(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) 
-	$(CC) -o $@ $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(LDLIBS)
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(LDLIBS) -o $@
 
 plot-xy-and-quad: plot-xy-and-quad.o matchfile.o $(COMMON)/cairoutils.o \
 		$(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) 
-	$(CC) -o $@ $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(LDLIBS)
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(LDLIBS) -o $@
 
 plot-constellations.o: plot-constellations.c
 	$(CC) $(CFLAGS) $(CAIRO_INC) $(NETPBM_INC) $(CATS_INC) -c -o $@ $<
 
 plot-constellations: plot-constellations.o $(CATS_LIB) $(COMMON)/cairoutils.o \
 		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) $(LDFLAGS) $^ -o $@ $(CAIRO_LIB) $(NETPBM_LIB) $(PNG_LIB) $(JPEG_LIB) $(LDLIBS)
+	$(CC) $(LDFLAGS) $^ $(CAIRO_LIB) $(NETPBM_LIB) $(PNG_LIB) $(JPEG_LIB) $(LDLIBS) -o $@
 
 catalog-dists: catalog-dists.o 2mass-fits.o usnob-fits.o usnob.o \
 		$(LIBKD_LIB) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
 
 add-text: add-text.o $(COMMON)/cairoutils.o $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) $(LDFLAGS) $^ -o $@ $(CAIRO_LIB) $(NETPBM_LIB) $(PNG_LIB) $(JPEG_LIB) $(LDLIBS)
+	$(CC) $(LDFLAGS) $^ $(CAIRO_LIB) $(NETPBM_LIB) $(PNG_LIB) $(JPEG_LIB) $(LDLIBS) -o $@
 
 PYTHON_EXECS := plotann.py
 PYTHON_INSTALL := $(PYTHON_EXECS)
@@ -645,7 +645,7 @@
 	$(CC) $(CFLAGS) $(CAIRO_INC) $(NETPBM_INC) -c -o $@ $<
 
 plotstuff: plotstuff-main.o $(PLOTSTUFF) $(CATS_LIB) $(COMMON)/cairoutils.o $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) -o $@ $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LDLIBS)
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LDLIBS) -o $@
 ALL_OBJ += plotstuff-main.o
 
 PLOTSTUFF_SRCS = $(subst .o,.c,$(PLOTSTUFF))
@@ -674,11 +674,11 @@
 
 plotxy: plotxy-main.o $(PLOTSTUFF) $(CATS_LIB) $(COMMON)/cairoutils.o \
 		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) -o $@ $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LDLIBS)
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LDLIBS) -o $@
 ALL_OBJ += plotxy-main.o
 
 test_plotstuff: test_plotstuff-main.o test_plotstuff.o $(COMMON)/cutest.o $(PLOTSTUFF) $(COMMON)/cairoutils.o $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) -o $@ $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(LDLIBS)
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(LDLIBS) -o $@
 test_plotstuff-main.c: test_plotstuff.c
 	$(AN_SHELL) $(MAKE_TESTS) $^ > $@
 
@@ -719,7 +719,7 @@
 ALL_OBJ += resort-xylist-main.o
 
 bgsubtract: bgsubtract.o ctmf.o $(COMMON)/cairoutils.o $(UTIL_OBJS) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) -o $@ $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(LDLIBS)
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(LDLIBS) -o $@
 
 usnob-scamp-catalog: usnob-scamp-catalog.o usnob-fits.o usnob.o $(UTIL_OBJS) \
 		$(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
diff -Nur astrometry.net-0.43.orig/util/Makefile astrometry.net-0.43/util/Makefile
--- astrometry.net-0.43.orig/util/Makefile	2013-02-18 07:12:36.000000000 -0800
+++ astrometry.net-0.43/util/Makefile	2013-06-13 17:05:10.551069687 -0700
@@ -181,7 +181,7 @@
 
 resample: resample-main.o \
 		$(ANFILES_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(WCS_LIB_DEP)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(WCS_LIB) $(LDLIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCS_LIB) $(LDLIBS) -o $@
 ALL_OBJ += resampe.o
 
 tycho2tofits: tycho2tofits.o \
@@ -195,10 +195,10 @@
 _sip.so: sip-utils.o gslutils.o sip.o starutil.o mathutil.o sip_qfits.o \
 		fitsioutils.o errors.o ioutils.o bl.o os-features.o an-endian.o \
 		log.o tic.o $(QFITS_LIB) $(GSL_LIB)
-	$(CC) $(SHAREDLIBFLAGS) -o $@ $^
+	$(CC) $(SHAREDLIBFLAGS) $^ $(GSL_LIB) -o $@
 
 _healpix.so: healpix-utils.o healpix.o permutedsort.o starutil.o mathutil.o errors.o ioutils.o bl.o log.o tic.o os-features.o
-	$(CC) $(SHAREDLIBFLAGS) -o $@ $^
+	$(CC) $(SHAREDLIBFLAGS) $^ -o $@
 
 util.py util_wrap.c: util.i $(ANUTILS_H)
 	swig -python -I. $<
@@ -225,22 +225,22 @@
 	cp build/lib/_index_util.so _index_util.so
 
 coadd: coadd-main.o $(ANUTILS_LIB) $(QFITS_LIB) $(GSL_LIB) $(WCS_LIB_DEP)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(WCS_LIB) $(LDLIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCS_LIB) $(LDLIBS) -o $@
 ALL_OBJ += coadd.o coadd-main.o
 
 wcs-resample: wcs-resample-main.o $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) $(WCS_LIB_DEP)
-	$(CC) -o $@ $^ $(LDFLAGS) $(WCS_LIB) $(LDLIBS)
+	$(CC) $^ $(LDFLAGS) $(WCS_LIB) $(LDLIBS) -o $@
 ALL_OBJ += wcs-resample-main.o
 
 fit-wcs: fit-wcs-main.o $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) $(WCS_LIB_DEP)
-	$(CC) -o $@ $^ $(LDFLAGS) $(WCS_LIB) $(LDLIBS)
+	$(CC) $^ $(LDFLAGS) $(WCS_LIB) $(LDLIBS) -o $@
 ALL_OBJ += fit-wcs-main.o
 
 get-healpix: get-healpix.o $(ANUTILS_LIB) $(GSL_LIB)
 ALL_OBJ += get-healpix.o
 
 wcs-to-tan: wcs-to-tan.o $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) $(WCS_LIB_DEP)
-	$(CC) -o $@ $^ $(LDFLAGS) $(WCS_LIB) $(LDLIBS)
+	$(CC) $^ $(LDFLAGS) $(WCS_LIB) $(LDLIBS) -o $@
 ALL_OBJ += wcs-to-tan.o
 
 pad-file: pad-file.o $(ANUTILS_LIB)
@@ -253,7 +253,7 @@
 ALL_OBJ += add-healpix-column.o
 
 an-pnmtofits: an-pnmtofits.o $(ANUTILS_LIB) $(QFITS_LIB)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(NETPBM_LIB) $(LDLIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(NETPBM_LIB) $(LDLIBS) -o $@
 ALL_OBJ += an-pnmtofits.o
 
 downsample-fits: downsample-fits.o $(ANUTILS_LIB) $(QFITS_LIB)
@@ -334,32 +334,32 @@
 	@echo
 	@echo "Testing canonicalize_file_name..."
 	@echo "Testing canonicalize_file_name..." >> os-features.log
-	(($(CC) -o os-features-test-canonicalize \
-	  $(CFLAGS) -DTEST_CANONICALIZE_FILE_NAME $^ $(LDFLAGS) >> os-features.log && \
+	(($(CC) $(LDFLAGS) \
+	  $(CFLAGS) -DTEST_CANONICALIZE_FILE_NAME $^ -o os-features-test-canonicalize >> os-features.log && \
 	  ./os-features-test-canonicalize >> os-features.log && \
 	  echo "#define NEED_CANONICALIZE_FILE_NAME 0") \
 	|| echo "#define NEED_CANONICALIZE_FILE_NAME 1") >> $@.tmp
 	@echo
 	@echo "Testing declare qsort_r..."
 	@echo "Testing declare qsort_r..." >> os-features.log
-	(($(CC) -o os-features-test-declare-qsort \
-	   $(CFLAGS) -DTEST_DECLARE_QSORT_R $^ $(LDFLAGS) >> os-features.log && \
+	(($(CC) $(LDFLAGS) \
+	   $(CFLAGS) -DTEST_DECLARE_QSORT_R $^ -o os-features-test-declare-qsort >> os-features.log && \
 	   ./os-features-test-declare-qsort >> os-features.log && \
 	   echo "#define NEED_DECLARE_QSORT_R 1") \
 	|| echo "#define NEED_DECLARE_QSORT_R 0") >> $@.tmp
 	@echo
 	@echo "Testing qsort_r..."
 	@echo "Testing qsort_r..." >> os-features.log
-	(($(CC) -o os-features-test-qsort \
-	   $(CFLAGS) -DTEST_QSORT_R $^ $(LDFLAGS) >> os-features.log && \
+	(($(CC) $(LDFLAGS) \
+	   $(CFLAGS) -DTEST_QSORT_R $^ -o os-features-test-qsort >> os-features.log && \
 	   ./os-features-test-qsort >> os-features.log && \
 	   echo "#define NEED_QSORT_R 0") \
 	|| echo "#define NEED_QSORT_R 1") >> $@.tmp
 	@echo
 	@echo "Testing swap qsort_r..."
 	@echo "Testing swap qsort_r..." >> os-features.log
-	(($(CC) -Werror -o os-features-test-swap-qsort \
-	   $(CFLAGS) -DTEST_SWAP_QSORT_R $^ $(LDFLAGS) >> os-features.log && \
+	(($(CC) $(LDFLAGS) -Werror \
+	   $(CFLAGS) -DTEST_SWAP_QSORT_R $^  -o os-features-test-swap-qsort >> os-features.log && \
 	   ./os-features-test-swap-qsort >> os-features.log && \
 	   echo "#define NEED_SWAP_QSORT_R 0") \
 	|| echo "#define NEED_SWAP_QSORT_R 1") >> $@.tmp
@@ -368,8 +368,8 @@
 	@echo "   NETPBM_INC is $(NETPBM_INC_ORIG)"
 	@echo "   NETPBM_LIB is $(NETPBM_LIB_ORIG)"
 	@echo "Testing netpbm..." >> os-features.log
-	(($(CC) -o os-features-test-netpbm \
-	   $(CFLAGS) -DTEST_NETPBM $(NETPBM_INC_ORIG) $^ $(LDFLAGS) $(NETPBM_LIB_ORIG) >> os-features.log && \
+	(($(CC) $(LDFLAGS) \
+	   $(CFLAGS) -DTEST_NETPBM $(NETPBM_INC_ORIG) $^ $(NETPBM_LIB_ORIG) -o os-features-test-netpbm >> os-features.log && \
 	   ./os-features-test-netpbm >> os-features.log && \
 	   echo "#define HAVE_NETPBM 1") \
 	|| echo "#define HAVE_NETPBM 0") >> $@.tmp
@@ -392,8 +392,8 @@
 	-rm -f os-features-makefile.log
 	( \
 	 echo "# This file is generated by util/Makefile."; \
-	 (($(CC) -o os-features-test-netpbm-make \
-	   $(CFLAGS) -DTEST_NETPBM_MAKE $(NETPBM_INC_ORIG) $^ $(LDFLAGS) $(NETPBM_LIB_ORIG) >> os-features-makefile.log && \
+	 (($(CC) $(LDFLAGS) \
+	   $(CFLAGS) -DTEST_NETPBM_MAKE $(NETPBM_INC_ORIG) $^ $(NETPBM_LIB_ORIG)  -o os-features-test-netpbm-make >> os-features-makefile.log && \
 	   ./os-features-test-netpbm-make >> os-features-makefile.log && \
 	   echo "HAVE_NETPBM := yes") \
 	|| (echo "# Astrometry.net didn't find netpbm; not setting HAVE_NETPBM."; \
@@ -502,10 +502,10 @@
 test_sip-utils: $(ANUTILS_LIB) $(QFITS_LIB) $(GSL_LIB)
 
 test_anwcs: $(ANUTILS_LIB) $(QFITS_LIB) $(GSL_LIB)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(WCSLIB_LIB)
+	$(CC)$(CFLAGS) $(LDFLAGS) $^ $(WCSLIB_LIB) -o $@
 
 test_wcs: $(ANFILES_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(GSL_LIB)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(WCSLIB_LIB)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCSLIB_LIB) -o $@
 
 test_tycho2: $(ANFILES_LIB) $(ANUTILS_LIB) $(QFITS_LIB)
 
diff -Nur astrometry.net-0.43.orig/util/makefile.tests astrometry.net-0.43/util/makefile.tests
--- astrometry.net-0.43.orig/util/makefile.tests	2010-07-16 12:48:05.000000000 -0700
+++ astrometry.net-0.43/util/makefile.tests	2013-06-13 16:24:33.816558534 -0700
@@ -50,5 +50,5 @@
 	$(AN_SHELL) $(MAKE_TESTS) $(TEST_SOURCES) > $@
 
 test: test.o $(COMMON)/cutest.o $(ALL_TEST_FILES_O) $(sort $(ALL_TEST_EXTRA_OBJS)) $(ALL_TEST_LIBS)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(ALL_TEST_EXTRA_LDFLAGS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(ALL_TEST_EXTRA_LDFLAGS) -o $@
 
